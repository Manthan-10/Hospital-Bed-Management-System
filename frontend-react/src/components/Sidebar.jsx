import React, { useState } from "react";
import Patient from "./Patient";

/**
 * This component displays all the Patients data in a Sidebar
 * @param {data} PatientDetails
 */
const Sidebar = ({ data }) => {
  const [searchTerm, setSearchTerm] = useState("");

  return (
    <nav className="sidebar">
      <div className="search-bar">
        <input
          type="text"
          className="search"
          placeholder="Search..."
          onChange={(event) => {
            setSearchTerm(event.target.value);
          }}
        />
      </div>
      {data &&
        data
          .filter((data) => {
            if (searchTerm === "") {
              return data;
            } else if (
              data.name.toLowerCase().includes(searchTerm.toLowerCase())
            ) {
              return data;
            }
          })
          .map((patient) => <Patient patient={patient} />)}
    </nav>
  );
};

export default Sidebar;
