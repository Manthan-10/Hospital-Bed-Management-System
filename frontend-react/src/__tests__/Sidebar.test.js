import React from 'react';
import Enzyme,{ shallow, mount } from 'enzyme';
import Sidebar from '../components/Sidebar';
import Adapter from 'enzyme-adapter-react-16';
import * as reactRouterDom from "react-router-dom";
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() })

/**
 * Testing for UI of Login page
 * @author MN097789
 */
 

 const mockNavigate = jest.fn();

 jest.mock("react-router-dom", () => ({
 
   useSelector: jest.fn(),
 
   useNavigate: () => mockNavigate,
 
 }));
 const mockNavigateSpy = jest.spyOn(reactRouterDom, "useNavigate");

 
 describe('Sidebar UI Test Suite', () => {

    let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Sidebar />);
  });

  test('Checks if it renders correctly', () => {
    const tree = renderer.create(<Sidebar/>).toJSON();
    expect(tree).toMatchSnapshot();
  });


  test('should check for divs', () => {
    wrapper = shallow(<Sidebar/>);
    expect(wrapper.length).toEqual(1);
  

    });
});