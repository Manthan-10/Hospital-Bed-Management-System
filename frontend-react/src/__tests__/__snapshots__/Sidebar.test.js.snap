// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar UI Test Suite Checks if it renders correctly 1`] = `
<nav
  className="sidebar"
>
  <div
    className="search-bar"
  >
    <input
      className="search"
      onChange={[Function]}
      placeholder="Search..."
      type="text"
    />
  </div>
</nav>
`;
